presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PruneUnreferencedOutputs.java

```diff
@@ -598,7 +598,7 @@ public PlanNode visitOutput(OutputNode node, RewriteContext<Set<VariableReferenc
         {
             Set<VariableReferenceExpression> expectedInputs = ImmutableSet.copyOf(node.getOutputVariables());
             PlanNode source = context.rewrite(node.getSource(), expectedInputs);
-            return new OutputNode(node.getSourceLocation(), node.getId(), source, node.getColumnNames(), node.getOutputVariables());
+            return new OutputNode(node.getSourceLocation(), node.getId(), node.getStatsEquivalentPlanNode(), source, node.getColumnNames(), node.getOutputVariables());
```

Comment:
@rschlussel this is an example of some rules that need to be changed to make this work. I did some minimal work to make tests pass, but more work needs to be done to fix these rules, and other rules.

Some things we can do to automate this:

* For IterativeOptimizer, we have pattern matches on topmost node that is rewritten. We can enforce/override that its StatsEquivalentPlanNode remains same.

* Perhaps we can do it for SimplePlanRewriter also, need to check correctness.